salt/modules/gpg.py

```diff
@@ -722,8 +723,8 @@ def import_key(text=None,
     if filename:
         try:
             with salt.utils.files.flopen(filename, 'rb') as _fp:
-                lines = _fp.readlines()
-                text = ''.join(lines)
+                text = ''.join([salt.utils.stringutils.to_unicode(x)
```

Comment:
I'm not sure why this was done this way originally, but if we don't need lines and we're reading the whole thing into memory anyway this would better just be

```suggestion
                text = salt.utils.stringutils.to_unicode(_fp.read())
```

No point in creating an intermediate list and then joining it if we just want the entire file read in as unicode.

In fact, I'm not familiar with the `flopen` function but I'm assuming that we can just pass it an encoding, so maybe it makes more sense to just do something like `flopen(filename, 'r', encoding='utf-8')` (or whatever it is that we're doing under the hood in `to_unicode`, but I'm pretty sure it's that)

Reply:
Used _to_unicode_ to allow for it working correctly under Windows (Shane's suggestion).

Against pulling apart function to see how they do it, and do it that way, better to use the function and pick up any issues that may be adjust or corrected later in the function. Similar for flopen given it will do different things on Solaris or AIX.

Reply:
That makes sense. 👍 for the `to_unicode` part, then, but we definitely shouldn't be doing the `''.join` bit. At least I don't see the need to create that intermediate list - `_fp.read()` should do the trick.